src/main/java/edu/harvard/iq/dataverse/authorization/groups/GroupServiceBean.java

```diff
@@ -103,6 +104,41 @@ public ShibGroupProvider getShibGroupProvider() {
     }
 
     /**
+     * groupsFor( RoleAssignee ra, DvObject dvo ) doesn't really get *all* the
+     * groups a Role assignee belongs to as advertised but this method comes
+     * closer.
+     *
+     * @todo Determine if this method works with IP Groups.
+     *
+     * @param au An AuthenticatedUser.
+     * @return As many groups as we can find for the AuthenticatedUser.
+     */
+    public Set<Group> groupsFor(AuthenticatedUser au) {
+        Set<Group> groups = new HashSet<>();
+        groups.addAll(groupsFor(au, null));
+        String identifier = au.getIdentifier();
+        String identifierWithoutAtSign = null;
+        if (identifier != null) {
+            try {
+                identifierWithoutAtSign = identifier.substring(1);
+            } catch (IndexOutOfBoundsException ex) {
+                logger.info("Couldn't trim first character (@ sign) from identifier: " + identifier);
+            }
+        }
+        if (identifierWithoutAtSign != null) {
+            roleAssigneeSvc.getUserGroups(identifierWithoutAtSign).stream().forEach((groupAlias) -> {
+                ExplicitGroup explicitGroup = explicitGroupProvider.get(groupAlias);
+                if (explicitGroup != null) {
```

Comment:
For the full lambda experience, you can use `map` to map the group alias to an explicit group, `filter` out the `null`s, and then use a collector to get the set of groups.